glfw/src/native.zig

```diff
@@ -0,0 +1,290 @@
+const std = @import("std");
+const c = @import("c.zig").c;
+
+const Window = @import("Window.zig");
+const Monitor = @import("Monitor.zig");
+const Error = @import("errors.zig").Error;
+const getError = @import("errors.zig").getError;
+
+const internal_debug = @import("internal_debug.zig");
+
+pub fn getWin32Adapter(monitor: Monitor) [*:0]const u8 {
+    internal_debug.assertInitialized();
+    const adapter = c.glfwGetWin32Adapter(monitor.handle);
+    getError() catch |err| return switch (err) {
+        Error.NotInitialized => unreachable,
+        else => unreachable,
+    };
+    return adapter;
+}
+
+pub fn getWin32Monitor(monitor: Monitor) [*:0]const u8 {
+    internal_debug.assertInitialized();
+    const mon = c.glfwWin32Monitor(monitor.handle);
+    getError() catch |err| return switch (err) {
+        Error.NotInitialized => unreachable,
+        else => unreachable,
+    };
+    return mon;
+}
+
+pub fn getWin32Window(window: Window) std.os.windows.HWND {
+    internal_debug.assertInitialized();
+    const win = c.glfwGetWin32Window(window.handle);
+    getError() catch |err| return switch (err) {
+        Error.NotInitialized => unreachable,
+        else => unreachable,
+    };
+    return @ptrCast(std.os.windows.HWND, win);
+}
+
+//
+
+pub fn getWGLContext(window: Window) std.os.windows.HGLRC {
+    internal_debug.assertInitialized();
+    const context = c.glfwGetWGLContext(window.handle);
+    getError() catch |err| return switch (err) {
+        Error.NotInitialized => unreachable,
+        Error.NoWindowContext => {}, // TODO
```

Comment:
We can/should fix these by returning an error set like this:

https://github.com/hexops/mach/blob/6866fce4ee3cb7c85f06adf603e416843725d60e/glfw/src/opengl.zig#L34

https://github.com/hexops/mach/blob/6866fce4ee3cb7c85f06adf603e416843725d60e/glfw/src/opengl.zig#L41

Reply:
I know, the PR is WIP :smile: